api/api/src/main/scala/uk/ac/wellcome/platform/api/QueryParams.scala

```diff
@@ -151,6 +154,9 @@ object MultipleWorksParams extends QueryParamsUtils {
   implicit val licenseFilter: Decoder[LicenseFilter] =
     decodeCommaSeparated.emap(strs => Right(LicenseFilter(strs)))
 
+  implicit val collectionsDepthFilter: Decoder[CollectionDepthFilter] =
+    decodeInt map CollectionDepthFilter
```

Comment:
I'm surprised this compiles, as used to getting errors when doing this and have to do `CollectionDepthFilter(_)` instead. Just checked apparently it only complains when you have a companion object defined as well:

```
[E]      type mismatch;
[E]       found   : uk.ac.wellcome.platform.api.models.CollectionDepthFilter.type
[E]       required: Int => uk.ac.wellcome.platform.api.models.CollectionDepthFilter
```

Reply:
Not 100% sure, I'll have a play around